be/src/pipeline/exec/olap_scan_operator.cpp

```diff
@@ -218,8 +218,40 @@ Status OlapScanLocalState::_should_push_down_function_filter(vectorized::Vectori
     return Status::OK();
 }
 
-bool OlapScanLocalState::_should_push_down_common_expr() {
-    return state()->enable_common_expr_pushdown() && _storage_no_merge();
+bool OlapScanLocalState::_should_push_down_common_expr(const vectorized::VExprSPtr& expr) {
+    if (!state()->enable_common_expr_pushdown()) {
+        return false;
+    }
+    if (_storage_no_merge() && vectorized::VExpr::is_acting_on_a_slot(*expr)) {
+        return true;
+    }
+    const auto& children = expr->children();
+    if (children.size() != 2) {
```

Comment:
why we need limit expr must has two child?

Reply:
only push down match predicate